middleware/nethttp/middleware.go

```diff
@@ -0,0 +1,91 @@
+package wasm
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"github.com/tetratelabs/wazero"
+)
+
+// Middleware is a factory of http.Handler instances implemented in Wasm.
+type Middleware interface {
+	// NewHandler creates a http.Handler implemented by a WebAssembly module.
+	// The returned handler will not invoke `next` when it obviates a request
+	// for reasons such as an authorization failure or serving from cache.
+	//
+	// ## Notes
+	//   - Each handler is independent, so they don't share memory.
+	//   - Handlers returned are not safe for concurrent use.
+	NewHandler(ctx context.Context, next http.Handler) (Handler, error)
+
+	// Close releases resources such as any Wasm modules, compiled code, and
+	// the runtime.
+	Close(context.Context) error
+}
+
+func NewMiddleware(ctx context.Context, guest []byte, options ...Option) (Middleware, error) {
+	o := &wasmOptions{
+		newRuntime:   DefaultRuntime,
+		moduleConfig: wazero.NewModuleConfig(),
+		logger:       func(msg string) {},
```

Comment:
Interesting one: I think sometimes you want to do the interpolation here to avoid interpolations when the logger is noop e.g. `m.logger(fmt.Sprintf("error: %v", err))`

Reply:
yeah per other thread very intentionally trying to avoid a logging bikeshed so quickly :D